packages/features/users/components/UserTable/BulkActions/MassAssignAttributes.tsx

```diff
@@ -241,78 +240,6 @@ function MassAssignAttributesBulkActionComponent({ table, filters }: Props) {
   const utils = trpc.useUtils();
   const bulkAssignAttributes = trpc.viewer.attributes.bulkAssignAttributes.useMutation({
     onSuccess: (success) => {
-      // Optimistically update the infinite query data
-      const selectedRows = table.getSelectedRowModel().flatRows;
-
-      utils.viewer.organizations.listMembers.setInfiniteData(
-        {
-          limit: 10,
-          searchTerm: "",
-          expand: ["attributes"],
-          filters,
-        },
-        // @ts-expect-error i really dont know how to type this
-        (oldData) => {
-          if (!oldData) {
-            return {
-              pages: [],
-              pageParams: [],
-            };
-          }
-
-          const newPages = oldData?.pages.map((page) => ({
-            ...page,
-            rows: page.rows.map((row) => {
-              if (selectedRows.some((selectedRow) => selectedRow.original.id === row.id)) {
-                // Update the attributes for the selected users
-
-                const attributeOptionValues = foundAttributeInCache?.options.filter((option) =>
-                  selectedAttributeOptions.includes(option.id)
-                );
-
-                const newAttributes =
-                  row.attributes?.filter((attr) => attr.attributeId !== selectedAttribute) || [];
-
-                if (attributeOptionValues && attributeOptionValues.length > 0) {
-                  const newAttributeValues = attributeOptionValues?.map((value) => ({
-                    id: value.id,
-                    attributeId: value.attributeId,
-                    value: value.value,
-                    slug: value.slug,
-                    contains: value.contains,
-                    isGroup: value.isGroup,
-                    weight: 100,
-                  }));
-                  newAttributes.push(...newAttributeValues);
-                } else {
-                  // Text or number input we don't have an option to fall back on
-                  newAttributes.push({
-                    id: "-1",
-                    attributeId: foundAttributeInCache?.id ?? "-1",
-                    value: selectedAttributeOptions[0],
-                    slug: slugify(selectedAttributeOptions[0]),
-                    contains: [],
-                    isGroup: false,
-                    weight: 100,
-                  });
-                }
-
-                return {
-                  ...row,
-                  attributes: newAttributes,
-                };
-              }
-              return row;
-            }),
-          }));
-
-          return {
-            ...oldData,
-            pages: newPages,
-          };
-        }
-      );
-
       setSelectedAttribute(undefined);
```

Comment:
I've removed optimistic updates in this file, and in another two files.

The existing implementation is very particular to each use case, and it takes quite a lot of effort. Even some of these were already broken.

So let's remove the optimistic updates in this PR, and see if it really needs further UX optimization. Then we can re-consider it.